### ISSUE TITLE
FlowNetS incorrect flow estimation in case of simple constant orizontal displacement

### ISSUE THREAD

**Issue by jeffbaena** [lmb-freiburg/dispnet-flownet-docker#11]
Dear authors,
 I am working on video coding and using FlowNetS for frame synthesis from optical flow.
Recently I have noticed that the optical flow estimated with FlowNetS was incorrect by a large margin on Monkaa.

To test the flow under controlled conditions I have generated the simple textures below below giving linear displacements to see how the estimation is.
Cases are:
Chess texture
- 100 px for x and 50 px for y. The flow estimated by flownetS is very far from the groundtruth
- 5 px only x. A bit better, but FlowNetS detects a displacement on y that don't exists

Face texture:
- 100px only x, delta y =0. Situation is a bit better than in chess texture, but also here there are a lot of artifacts and a y displacement that is not present on groundtruth
- 5 px only x. Situation is generally better

Double check: zero displacement
- uncertainty of 1 px, acceptable.

**Are these results reliable to you?**  A linear constant displament is the most simple case of motion and it should be detected perfectly. Is there any explanation you can give on these results?

Kind Regards,
Stefano

Large displacement texture
![ld_texture](https://user-images.githubusercontent.com/18122010/50007601-c093b500-ffb1-11e8-8873-c8e07afa3d1c.png)
Large Displacement Flow
![ld_texture_flo_axis](https://user-images.githubusercontent.com/18122010/50010235-20418e80-ffb9-11e8-9167-3574e019fb56.png)
Small displacement texture
![sd_texture](https://user-images.githubusercontent.com/18122010/50007629-da34fc80-ffb1-11e8-89e7-12edcefa8b50.png)
Small displacement flow
![sd_texture_flo_axis](https://user-images.githubusercontent.com/18122010/50010259-36e7e580-ffb9-11e8-8ac1-b2e95f59a1e4.png)

Face LD
![ld_face_texture](https://user-images.githubusercontent.com/18122010/50007655-eae57280-ffb1-11e8-8865-42b0eee06369.png)
Flo face LD
![ld_face_flo_axis](https://user-images.githubusercontent.com/18122010/50010285-46ffc500-ffb9-11e8-86f3-349b61b147e4.png)

Face SD
![sd_face_texture](https://user-images.githubusercontent.com/18122010/50010296-4f580000-ffb9-11e8-88d1-c50a1c79f4cb.png)
Face flo SD
![sm_face_flo_axis](https://user-images.githubusercontent.com/18122010/50010316-5da61c00-ffb9-11e8-833b-a6f616126820.png)

**Comment by nikolausmayer** [comment]
Hi,
nice tests! While "traditional" flow methods usually perform better on smaller motions and cases which lend themselves well to inter/extrapolation (such as your checkerboard), our supervisedly trained FlowNets have problems with data which differs from the statistics presented during training. And most of our training data is well textured with large motions and many objects moving independently. 
And FlowNetS is our oldest and "worst" FlowNet. If you have the compute capacity to run any other FlowNet, there's no good reason to use FlowNetS :slightly_smiling_face: 

For your SD cases, I would suggest you run "FlowNet2-SD"  (available in another Docker image: https://github.com/lmb-freiburg/flownet2-docker), or the newer "FlowNet3/CSS-ft-sd" (available in https://github.com/lmb-freiburg/netdef-docker). 
These networks have been trained or finetuned using our [ChairsSDHom](https://lmb.informatik.uni-freiburg.de/resources/datasets/FlyingChairs.en.html#chairssdhom) dataset which focuses on small motions and also teaches untextured regions.

For your large-displacement cases, I would test either "FlowNet2" (found in "flownet2-docker") or "FlowNet3-CSS" ("netdef-docker").

The "netdef-docker" image runs TensorFlow, with the advantage that even a relatively small GPU can run large networks (just slower). "flownet2-docker" uses Caffe which runs into hard memory limits earlier.

However, the case where the entire image has the same flow vector is a very unnatural setting (although I do understand that it is very relevant in video flow compression). I think none of our networks will perform terribly well on this; you should always expect some noise in the output. For these cases, traditional methods might work better.

**Comment by jeffbaena** [comment]
Dear Nikolaus,
 thanks so much for your prompt answer and explanations!

I will definitely try the others FlowNet, however FlowNetS was really appealing to me as it is simpler and can easily being fine tuned on other datasets. I think that FlowNet2 finetuning might be a bit more cumbersome due to the many stacked networks.

Thanks again for your help,
Stefano
